yt_dlp/extractor/mlb.py

```diff
@@ -263,3 +265,68 @@ def _download_video_data(self, display_id):
   }
 }''' % display_id,
             })['data']['mediaPlayback'][0]
+
+
+class MLBTVIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?mlb\.com/tv/g(?P<id>\d{6})/?.*'
+
+    _TESTS = [{
+        'url': 'https://www.mlb.com/tv/g661581/vee2eff5f-a7df-4c20-bdb4-7b926fa12638',
+        'info_dict': {
+            'id': '661581',
+            'ext': 'mp4',
+            'title': '2022-07-02 - St. Louis Cardinals @ Philadelphia Phillies',
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
+    def _perform_login(self, username, password):
+
+        access_token = self._download_json('https://ids.mlb.com/oauth2/aus1m088yK07noBfh356/v1/token', None,
+                                           headers={'User-Agent': 'okhttp/3.12.1',
+                                                    'Content-Type': 'application/x-www-form-urlencoded'},
+                                           data=(('grant_type=password&username=%s&password=%s&scope=openid offline_access'
+                                                  '&client_id=0oa3e1nutA1HLzAKG356') % (urllib.parse.quote(username), urllib.parse.quote(password)))
+                                           .encode())['access_token']
+
+        entitlement = self._download_webpage('https://media-entitlement.mlb.com/api/v3/jwt?os=Android&appname=AtBat&did=' + str(uuid.uuid4()), None,
```

Comment:
I think `os=Android&appname=...` part can moved to `query` field

Reply:
I am not quite sure what you mean here... is `query` preferred to just using the URL?